Sources/Crypto/Cipher/AuthenticatedCipher.swift

```diff
@@ -0,0 +1,132 @@
+import CNIOOpenSSL
+import Foundation
+import Bits
+
+/// AES-256 GCM Cipher. This is the reccomended cipher mode. (see https://github.com/vapor/crypto/issues/59).
+///
+///     let key: Data // 32 bytes
+///     let iv: Data // 12 RANDOM bytes; different for each plaintext to encrypt. MUST be passed alongside the ciphertext to the receiver.
+///     let (ciphertext, tag) = try AES256GCM.encrypt("vapor", key: key, iv: iv)
+///     print(ciphertext) // Encrypted Data
+///     AES256GCM.decrypt(ciphertext, key: key, iv: iv, tag: tag).convert(to: String.self) // "vapor"
+///
+public var AES256GCM: AuthenticatedCipher { return .init(algorithm: .aes256gcm) }
+
+public final class AuthenticatedCipher: OpenSSLStreamCipher {
+    /// The `AuthenticatedCipherAlgorithm` (e.g., AES-256-GCM) to use.
+    public let algorithm: OpenSSLCipherAlgorithm
+
+    /// Internal OpenSSL `EVP_CIPHER_CTX` context.
+    public let ctx: UnsafeMutablePointer<EVP_CIPHER_CTX>
+
+    /// Byte length of a GCM tag
+    public static let gcmTagLength: Int = 16
```

Comment:
This is specific to AES-GCM, while the rest of the class isn't. Would there be a way around that?